enterprise/server/backends/distributed/distributed.go

```diff
@@ -142,12 +177,50 @@ func (c *Cache) GetMulti(ctx context.Context, digests []*repb.Digest) (map[*repb
 	return foundMap, nil
 }
 
-func (c *Cache) Set(ctx context.Context, d *repb.Digest, data []byte) error {
-	peer := c.peer(d)
-	if peer == c.myAddr {
-		return c.local.Set(ctx, d, data)
+type multiCloser struct {
+	closers []io.WriteCloser
+}
+
+func (mc *multiCloser) Write(data []byte) (int, error) {
+	for _, w := range mc.closers {
+		n, err := w.Write(data)
```

Comment:
do you want to do these in parallel? (same w/ `Close()` below)

Reply:
I'm fine starting here -- don't want to prematurely spend time on that if this thing isn't workable anyway.